src/win32.c

```diff
@@ -1032,4 +1032,35 @@ gchar *win32_get_user_config_dir(void)
 	return g_build_filename(g_get_user_config_dir(), "geany", NULL);
 }
 
+
+/* Retrieve the console codepage
+ * In case GetConsoleOutputCP() returns 0 (i.e. the application doesn't have an own console window
+ * fallback to GetOEMCP(). */
+guint win32_get_console_codepage(void)
+{
+	guint codepage = GetConsoleOutputCP();
+	if (codepage == 0)
+		codepage = GetOEMCP();
+	return codepage;
+}
+
+
+/* Convert a string into the system's default codepage, this is different from the
+ * locale (e.g. default codepage is 850 but locale is CP1252).
+ * This assumes the input string is encoded as UTF-8, otherwise a copy of
+ * the input string is returned. */
+#define CODEPAGE_LEN 8
```

Comment:
instead of this, you could just declare the array `gchar codepage[8] = { 0 };` and then use `G_N_ELEMENTS(codepage)`.  Removes the (local) define that seem here just to avoid writing the magic `8` twice rather than something more global